### DISCUSSION TITLE
Driver selftest

### DISCUSSION THREAD

**Discussion by MohanPrasad-J** [linux-netdev/nipa#26] [Q&A]
Hi, I'm trying to add a self test for my network driver. I have the test running locally. Now i want to add it to the NetDev CI system such that the test gets executed automatically on every branch event. The documentation here is not so helpful. Could someone provide the steps to achieve this. 
1. Specifically when i tried to push the code to testing repository i get "permission denied"error.  Who can give access? or how to request for access?
2. Is the testing repository the right repository for adding tests that should go to the net-next branch ?

**Comment by kuba-moo** [comment]
This repo holds scripts used by upstream Linux maintainers. If you're working upstream the tests should be sent with other upstream patches. If you're not working upstream this repo is completely useless to you.
